[ZEPPELIN-996] Improve first page and dropdown menu in documentation site

AhyoungRyu opened the pull request:
### What is this PR for?
Current Zeppelin documentation site is little bit hard to find a way for Zeppelin beginners. It will not easy  to improve this at a time, but I did the below as a start of this work.

1. Restructured dropdown menu and added each category names
2. Added a overview list(with short description) to first page of website (index.md) so that people can look through the overall contents in Zeppelin website at a glance. (as [Apache Spark](http://spark.apache.org/docs/latest/#where-to-go-from-here) and [Apache Mesos](http://mesos.apache.org/documentation/latest/) does)

Please see the attached screenshot images :)

### What type of PR is it?
Improvement & Documentation

### Todos
* [x] - Change outdated screenshot images 
* [x] - Combine `text.md`, `table.md` and `html.md` to `basicdisplaysystem.md`
* [x] - Fix dead link in `virtual_machine.md`
* [x] - Improve dropdown menu and reorder
* [x] - Improve first page(`index.md`)
* [x] - Combine with #995 after it is merged into master

### What is the Jira issue?
[ZEPPELIN-996](https://issues.apache.org/jira/browse/ZEPPELIN-996)

### How should this be tested?
1. Apply this patch and [build the docs website with jekyll](https://github.com/apache/zeppelin/tree/master/docs#build-documentation)
2. Check the first page(index.html) and dropdown menu

### Screenshots (if appropriate)
 - Dropdown menu
![dropdown](https://cloud.githubusercontent.com/assets/10060731/16061421/b44f8034-3241-11e6-88fd-43aa5031b453.gif)


 - First page
<img width="717" alt="screen shot 2016-06-14 at 1 28 58 pm" src="https://cloud.githubusercontent.com/assets/10060731/16058631/3ab2cb6c-3234-11e6-95f4-180290df3d02.png">
<img width="694" alt="screen shot 2016-06-14 at 1 29 11 pm" src="https://cloud.githubusercontent.com/assets/10060731/16058639/43d68918-3234-11e6-921c-28436bfca33d.png">
<img width="649" alt="screen shot 2016-06-14 at 1 29 39 pm" src="https://cloud.githubusercontent.com/assets/10060731/16058650/501ec6d6-3234-11e6-9292-53ae84acc18a.png">
<img width="684" alt="screen shot 2016-06-14 at 1 29 57 pm" src="https://cloud.githubusercontent.com/assets/10060731/16058643/4637c8f2-3234-11e6-9b12-a233906f4c8b.png">
<img width="650" alt="screen shot 2016-06-14 at 1 30 12 pm" src="https://cloud.githubusercontent.com/assets/10060731/16058655/56c5af22-3234-11e6-8d29-9b7937728948.png">
<img width="636" alt="screen shot 2016-06-14 at 1 30 22 pm" src="https://cloud.githubusercontent.com/assets/10060731/16058656/58d1187e-3234-11e6-9171-ab7390b4a526.png">




### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

corneadoug commented:
Great work,
Maybe in the case of the interpreter menu, we should bring the usage section before the available interpeters (since there is only one item)

AhyoungRyu commented:
@corneadoug Ah there is two menu actually, they are not shown properly in the screenshot. 
 - Dynamic Interpreter Loading 
 - Dependency Management 

Anyway, I can change the placement as you suggested :)

corneadoug commented:
Oh indeed,
Since they are more generalistic and also a but hidden down there, it could be a better read flow to have them right below the overview

AhyoungRyu commented:
@corneadoug Yeah you're right and it makes sense indeed. I addressed it :)

bzz commented on docs/displaysystem/basicdisplaysystem.md:
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
 {% include JB/setup %}
+## Html
+
+With `%html` directive, Zeppelin treats your output as html
```
A nitpick - it's probably 'as HTML' capital case, as it's referred below

bzz commented on docs/displaysystem/basicdisplaysystem.md:
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
+
+## Text
+
+Zeppelin prints output of language backend in text, by default.
```
By default, Apache Zeppelin prints interpreter responce as a plain text using `text` display system.

bzz commented on docs/displaysystem/basicdisplaysystem.md:
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
+
+<img src="/assets/themes/zeppelin/img/screenshots/display_text.png" />
+
+You can explicitly say you're using text display system.
```
It is possible to explicitly specify display system as `text`

bzz replied:
It is possible to explicitly specify display system as `text` as well

bzz replied:
It is also possible to explicitly specify display system as `text`

bzz commented on docs/displaysystem/basicdisplaysystem.md:
```diff
@@ -18,19 +18,36 @@ See the License for the specific language governing permissions and
+
+<img src="/assets/themes/zeppelin/img/screenshots/display_text1.png" />
+
+Note that display system is backend independent.
```
I think exactly the same sentence is already present above

AhyoungRyu replied:
Thanks for pointing out them. I should've read overall contents again :)

bzz commented on docs/index.md:
```diff
@@ -74,49 +71,104 @@ Some basic charts are already included in Zeppelin. Visualizations are not limit
+## Dynamic forms
 
-Zeppelin can dynamically create some input forms into your notebook.
+You can dynamically create some input forms in your notebook.
```
Generally, previous writing style of "Zeppelin can ..." is prefered to the one that is dominant in this PR, like "You can ...", "Your notebooks ..." - first one feels more like a documentation about the project.

What do you think, is it possible\make sense to adjust it here?

AhyoungRyu replied:
Sure why not. It makes sense indeed. Thanks!

AhyoungRyu commented:
@bzz Thanks for your prudent review! I just addressed them in [6a2f40b](https://github.com/apache/zeppelin/pull/1004/commits/6a2f40b73e65986bdd61b611c76243f7a8bd7916).

minahlee commented:
<img width="1280" alt="screen shot 2016-06-14 at 10 34 51 am" src="https://cloud.githubusercontent.com/assets/8503346/16052985/ad3fa0ec-321b-11e6-9f15-e9725cee700f.png">

This is what I see from my screen, and cannot scroll down lower. Can you adjust size of the dropdown menu?

And also, can we rollback the change of display system menu order to text -> html -> table? if there are a lot of menu then alphabetical order makes sense, but in this case I think better to display in order of basic to advanced

Other than above comments, this PR looks awesome!

AhyoungRyu commented:
@minahlee Thanks for your prudent feedback. I just addressed them to my last some commits as you said. And I also applied scrollbar to "More" tab. 

BTW, I found an UI issue on "More" tab. When the screen width is small, some long text in the list are not visible. It was also fixed at [af70939](https://github.com/apache/zeppelin/pull/1004/commits/af70939b4231101e7f3760b6367a05704d3912c4)

 - Before 
![screen shot 2016-06-14 at 1 36 10 pm](https://cloud.githubusercontent.com/assets/10060731/16058906/7bba0570-3235-11e6-84a5-73d97ede5086.png)

 - After
![screen shot 2016-06-14 at 1 36 58 pm](https://cloud.githubusercontent.com/assets/10060731/16058909/7fbc7d2e-3235-11e6-9ab7-426f768e7ce8.png)

minahlee commented:
merging if there is no more discussion
